### PR TITLE
feat: initialize project layout

### PR DESCRIPTION
**Pull request by Hidayathamir**
I initialize project layout, combining [brix](https://github.com/brix-go/brix), [golang-standards](https://github.com/golang-standards/project-layout) and our current folder structure in eraspace.

## How to Run

1. Rename `.env-example` to `.env`.

```shell
mv .env-example .env
```

2. Run database using docker compose in folder `deployment`.

```shell
cd deployment && sudo docker compose up
```

3. Run `main.go` in other terminal.

```shell
go run main.go
```

4. Test dummy API.
```shell
curl --location 'http://localhost:4000/api/v1/baz/true'
```
```shell
curl --location 'http://localhost:4000/api/v1/baz/false'
```

### CODE REVIEW

File: external/README.md
Comment (irvankadhafi) [inline external/README.md:1]: ini perlu di bahas, kita perlu ngebuat client nya di penyedia grpc/http 

referensi: https://github.com/marselester/ddd-err/blob/master/apiclient/client.go

File: internal/api/v1/cart/repository/repository.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package repository
+
+import (
+	"errors"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type ICartRepository interface {
```
Comment (irvankadhafi) [inline internal/api/v1/cart/repository/repository.go:10]: perlu ada convention mengenai naming interface

File: internal/api/v1/cart/repository/repository.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+type cartRepository struct {
+	db *gorm.DB
+}
+
+func NewCartRepository(db *gorm.DB) ICartRepository {
+	return &cartRepository{db: db}
+}
+
+func (c *cartRepository) Foo(ctx *gin.Context) (string, error) {
```
Comment (irvankadhafi) [inline internal/api/v1/cart/repository/repository.go]: gin.Context better di level delivery/controller aja

File: external/README.md
Comment (Hidayathamir) [reply external/README.md:1]: Oke, kita bahas next meeting.

cc mas @alanpryoga

File: internal/api/v1/cart/repository/repository.go (hunk 1 above)
Comment (Hidayathamir) [reply internal/api/v1/cart/repository/repository.go:10]: Oke kita bahas next meeting.

cc mas @alanpryoga

File: internal/api/v1/cart/repository/repository.go (hunk 2 above)
Comment (Hidayathamir) [reply internal/api/v1/cart/repository/repository.go]: agree, gin.Context hanya untuk layer controller.

layer service & repository pakai context.Context.

thanks for the reminder, will update it.

Comment (Hidayathamir) [reply internal/api/v1/cart/repository/repository.go]: fixed here https://github.com/spacetronot-research-team/cart-service/pull/2/commits/e8e078fdb0be1b66c00e4256ad00243d357dbd7c

**Pull request closed without merging**